trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -140,7 +143,7 @@ public void proxyRequest(SigningMetadata signingMetadata, ParsedS3Request reques
 
         request.requestContent().contentLength().ifPresent(length -> remoteRequestHeadersBuilder.putOrReplaceSingle("content-length", Integer.toString(length)));
 
-        contentInputStream(request.requestContent(), signingMetadata).ifPresent(inputStream -> remoteRequestBuilder.setBodyGenerator(streamingBodyGenerator(inputStream)));
+        contentInputStream(request.requestContent(), signingMetadata).ifPresent(inputStream -> remoteRequestBuilder.setBodyGenerator(streamingBodyGenerator(quotaStreamController.wrap(inputStream))));
```

Comment:
I don't think this is the right stream to wrap. We want to limit the upper bound of the number of bytes we could ever read from the client, so I think we should be wrapping this here: https://github.com/trinodb/aws-proxy/blob/d95612067440af5a92ad5c28041883fff2bb7eb0/trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/TrinoS3ProxyClient.java#L185-L200

Otherwise, it is possible to defeat the purpose of this PR by sending an `aws-chunked` request with content like:

```
1,chunk-signature=<very long string>\r\n
f
```

The `AwsChunkedInputStream` will only ever "read" 1 byte, despite the actual payload being read in is much much larger.

If we wrap the input stream from the request, we would prevent this corner case. We get into a potential pitfall where users need to keep in mind that the quota needs to account for the overhead of aws chunking headers.

We can alternatively make `AwsChunkedInputStream` smarter and ensure it never reads more bytes than it should (we know the size of a signature, so computing an upper bound for the size of a chunk header should be trivial)

Reply:
done